libtcmu.c

```diff
@@ -277,78 +268,43 @@ static int generic_handle_cmd(struct tcmu_device *dev,
 	case MODE_SELECT:
 	case MODE_SELECT_10:
 		return tcmu_emulate_mode_select(cdb, iovec, iov_cnt, sense);
+	default:
+		tcmu_err("unknown command %x\n", cdb[0]);
+		return TCMU_NOT_HANDLED;
+	}
+}
+
+static bool command_is_store_call(uint8_t cmd)
+{
```

Comment:
There's no hard rule to "classify" commands as store and non-store calls, but a simple rule that I used here was - if a command needs to invoke handler calls which possibly would end up performing backend IO, these commands can be invoked via worker thread(s). Such classification is what's used in this PR to group commands as "store" calls. Non store calls are finished in the callers (command processing thread) context.

This is really not replacing the passthrough mechanism by any means. We'd need to see how PR #91 possibly benefit from this change. That said, I've not been following PR #91 closely and it has changed a bit from the last time I looked at it.